tests/test_dimensions.py

```diff
@@ -3,67 +3,95 @@
 
-def test_units():
-    d1 = ansunits.Dimensions(units="ft")
-    assert d1.units == "ft"
-    assert d1.dimensions == [0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
+def test_dimensions():
+    d1 = ansunits.Dimensions(
+        dimensions_container=[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
+    )
+    assert d1.full_list == [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
 
-    d2 = ansunits.Dimensions(units="kPa")
-    assert d2.units == "kPa"
-    assert d2.dimensions == [1.0, -1.0, -2.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
+    d2 = ansunits.Dimensions(dimensions_container=[0, 1.0, -2.0])
+    assert d2.full_list == [0.0, 1.0, -2.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
 
-    d3 = ansunits.Dimensions(units="slug^2 ft^-1 s R radian^-3 slugmol cd^4 A sr")
-    assert d3.units == "slug^2 ft^-1 s R radian^-3 slugmol cd^4 A sr"
-    assert d3.dimensions == [2.0, -1.0, 1.0, 1.0, -3.0, 1.0, 4.0, 1.0, 1.0]
+    d3 = ansunits.Dimensions(
+        dimensions_container=[1.0, -1.0, -2.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
+    )
+    assert d3.full_list == [1.0, -1.0, -2.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
 
 
-def test_dimensions():
-    d1 = ansunits.Dimensions(dimensions=[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0])
-    assert d1.units == ""
-    assert d1.dimensions == [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
+def test_dictionary():
+    d1 = ansunits.Dimensions(dimensions_container=[0, 1.0, -2.0])
+    assert d1.short_dictionary == {"length": 1.0, "time": -2.0}
+    d2 = ansunits.Dimensions(
+        dimensions_container=[1.0, 0.0, -2.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0]
+    )
+    assert d2.short_dictionary == {"mass": 1, "time": -2.0, "light": 1}
 
-    d2 = ansunits.Dimensions(dimensions=[0, 1.0, -2.0])
-    assert d2.units == "m s^-2"
-    assert d2.dimensions == [0.0, 1.0, -2.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
 
-    d3 = ansunits.Dimensions(dimensions=[1.0, -1.0, -2.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0])
-    assert d3.units == "kg m^-1 s^-2"
-    assert d3.dimensions == [1.0, -1.0, -2.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
+def test_short_list():
+    d1 = ansunits.Dimensions(dimensions_container=[0, 1.0, -2.0])
+    assert d1.short_list == [0, 1.0, -2.0]
+    d2 = ansunits.Dimensions(
+        dimensions_container=[1.0, 0.0, -2.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0]
+    )
+    assert d2.short_list == [1.0, 0.0, -2.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0]
 
 
-def test_unit_sys():
+def test_max_dim_len():
+    max_len = ansunits.Dimensions.max_dim_len()
+    assert max_len == 10
+
+
+def test_str_():
     d1 = ansunits.Dimensions(
-        dimensions=[1.0, -1.0, -2.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
-        unit_sys=["slug", "ft", "s", "R", "radian", "slugmol", "cd", "A", "sr"],
+        dimensions_container=[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0]
     )
-    assert d1.units == "slug ft^-1 s^-2"
-    assert d1.dimensions == [1.0, -1.0, -2.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
+    assert d1.__str__() == "[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0]"
```

Comment:
It's better to use `assert str(d1) == ...`.  There's a reason to add the dunder method :-)

Reply:
That is much cleaner thank you.